docs/source/en/using-diffusers/other-formats.md

```diff
@@ -10,167 +10,218 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
+There are two ways model files are organized, either in multiple folders or in a single file. Diffusers uses the multifolder layout by default where each component file (text encoder, UNet, VAE) is stored in a separate subfolder. Diffusers also supports loading models from a single file where all the components are bundled together.
 
-The Convert KerasCV Space allows you to input the following:
+### Multifolder
 
-* Your Hugging Face token.
-* Paths to download the UNet and text encoder weights from. Depending on how the model was trained, you don't necessarily need to provide the paths to both the UNet and text encoder. For example, Textual Inversion only requires the embeddings from the text encoder and a text-to-image model only requires the UNet weights.
-* Placeholder token is only applicable for textual inversion models.
-* The `output_repo_prefix` is the name of the repository where the converted model is stored.
+The multifolder layout is the default storage layout for Diffusers. Each component's (text encoder, UNet, VAE) weights are stored in a separate subfolder. The weights can be stored as any file type such as `.safetensors` or `.ckpt`.
 
-Click the **Submit** button to automatically convert the KerasCV checkpoint! Once the checkpoint is successfully converted, you'll see a link to the new repository containing the converted checkpoint. Follow the link to the new repository, and you'll see the Convert KerasCV Space generated a model card with an inference widget to try out the converted model.
+<div class="flex flex-row gap-4">
+  <div class="flex-1">
+    <img class="rounded-xl" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/multifolder-layout.png"/>
+    <figcaption class="mt-2 text-center text-sm text-gray-500">multifolder layout</figcaption>
+  </div>
+  <div class="flex-1">
+    <img class="rounded-xl" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/multifolder-unet.png"/>
+    <figcaption class="mt-2 text-center text-sm text-gray-500">UNet subfolder</figcaption>
+  </div>
+</div>
 
-If you prefer to run inference with code, click on the **Use in Diffusers** button in the upper right corner of the model card to copy and paste the code snippet:
+To load from a multifolder layout, use the [`~DiffusionPipeline.from_pretrained`] method.
 
 ```py
 from diffusers import DiffusionPipeline
 
 pipeline = DiffusionPipeline.from_pretrained(
-    "sayakpaul/textual-inversion-cat-kerascv_sd_diffusers_pipeline", use_safetensors=True
-)
+    "stabilityai/stable-diffusion-xl-base-1.0",
+    torch_dtype=torch.float16,
+    variant="fp16",
+    use_safetensors=True,
+).to("cuda")
 ```
 
-Then, you can generate an image like:
+Benefits of using the multifolder layout include:
 
-```py
-from diffusers import DiffusionPipeline
+1. Faster to load each component file individually or in parallel.
```

Comment:
diffusers-style format provides much greater flexibility and allows users to customize loading. We should expand on this section and support it with more practical and convincing examples! (the example here demonstrates the API they can use to customize loading, but it is very simple and it does not directly demonstrate the speed and memory-saving advantage, users have to deduce that themselves)

if you want to use two different models in single_file, most likely they will share some components, e.g. vae, text_encoders etc, and you will need to load these models twice - it's slower and they take up more memories than needed.  With diffusers-format, you would load one model, and load only the different components for the second one, and use the `from_pipe` API to switch.  that  cc @asomoza here. Can you help think of an example that will be meaningful for the single-format fans? 

another strong argument we can make is when the model includes large components - I wonder how it is even possible for them to handle models like sd3 in a single-file format? with 3 text encoders (including t5) and a 8b unet? also cc @asomoza here for insights -

Reply:
It's mostly the same that you mentioned, here are the "maybe" reasons:

- For SDXL we can literally use just one [vae](https://huggingface.co/madebyollin/sdxl-vae-fp16-fix) for all the models, so if you have 10 models, you will save 3.5 GB of space and bandwidth.
- For SD 1.5 there's like 4 or 5 good vae models, they're smaller but even with that, since we can switch them on the fly, we can always use the optimal one for each model.

So all the models could be distributed without vaes, in sites like civitai.com there's always a confusion with models that come with or without vae, this is not a problem with the diffusers format.

For the SDXL and SD 1.5 text encoders, we could save that space, but we really don't know which models have trained the text encoders or not, so this is harder to make evident to the single-file fans unless the model owners and sharing sites start to make that information visible.

These ones are the "strong" reasons:

- Turbo, Lighting and Hyper-SD models, these ones just change the unet, so there's no reason to share anything else.
- New models with T5 (Pixart Sigma, Kandinsky 3, SD3), the text encoder alone has a size between 17GB and 22 GB, It’s simply not feasible to share the text encoder with every fine-tuned model, and probably no one will train them either. There’s really no need to do so.

IMO the last one is the main reason people will be forced to stop sharing single-file checkpoints.